tests/scripts/multus-test.sh

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env bash
+set -eEuo pipefail
+
+##
+## This script will install a configurable number of multus-networked daemonsets to test that your
+## multus configuration and network hardware can provide enough IP addresses for Ceph.
+## Configuration options are commented below.
+##
+## TODO: in the future, this test should also check network stability under a reasonable amount of
+##       load to/from daemons.
+##
+
+## A comma-delimited list of multus networks to use for test pods
+: "${MULTUS_NETWORKS:="public-net,cluster-net"}"
+
+## The number of daemons to run per node as a test. This should reflect the overall size expected
+## for your Ceph install. As an easy guideline, add a daemon for each additional OSD beyond 3.
+: "${NUM_DAEMONS_PER_NODE:=15}"
+
+## The namespace to install multus test daemons into.
+: "${VALIDATION_NAMESPACE:="rook-ceph"}"
+
+## If you use a command other than 'kubectl' for getting/creating cluster resources, change this.
+: "${KUBECTL:="kubectl"}"
+
+## If you use a command other than 'jq' for json querying, change this.
+: "${JQ:="jq"}"
+
+ALL_DAEMONSETS=() # global for cleanup function
+function main() {
+  trap cleanup EXIT
+
+  ALL_DAEMONSETS=()
+  for i in $(seq "$NUM_DAEMONS_PER_NODE"); do
+    daemonset_name="multus-validator-${i}"
+    ALL_DAEMONSETS+=("${daemonset_name}")
+
+    render_daemonset "${daemonset_name}" | $KUBECTL create -f -
+  done
+
+  echo "Waiting for ${#ALL_DAEMONSETS[@]} daemonsets to have pods scheduled"
+  num_expected_pods=0
+  for ds in "${ALL_DAEMONSETS[@]}"; do
+    echo " └─ waiting for daemonset '${ds}' to have pods scheduled"
+    while true; do
+      num_desired="$($KUBECTL --namespace $VALIDATION_NAMESPACE get daemonset "${ds}" \
+        --output jsonpath='{.status.desiredNumberScheduled}' || true)"
+      if [[ -n "$num_desired" ]] && [[ "$num_desired" -gt 0 ]]; then
+        num_expected_pods=$((num_expected_pods + num_desired))
+        break
+      fi
+      sleep 1
+    done
+  done
+
+  # read -p "wait for input..."
```

Comment:
```suggestion
  # read -p "wait for input..."
```
do we need this comment?

Reply:
Oops :+1: